keel-web/src/main/kotlin/com/netflix/spinnaker/keel/rest/ApplicationController.kt

```diff
@@ -75,13 +75,17 @@ class ApplicationController(
       if (includeDetails && !entities.contains("resources")) {
         entities.add("resources")
       }
-      entities.forEach { entity ->
-        results[entity] = when (entity) {
-          "resources" -> applicationService.getResourceSummariesFor(application)
-          "environments" -> applicationService.getEnvironmentSummariesFor(application)
-          "artifacts" -> applicationService.getArtifactSummariesFor(application,
-            maxArtifactVersions ?: DEFAULT_MAX_ARTIFACT_VERSIONS)
-          else -> throw InvalidRequestException("Unknown entity type: $entity")
+      if (entities.size == 3) { // requesting everything, can optimize
+        results.putAll(applicationService.getSummariesAllEntities(application))
+      } else {
+        entities.forEach { entity ->
+          results[entity] = when (entity) {
+            "resources" -> applicationService.getResourceSummariesFor(application)
+            "environments" -> applicationService.getEnvironmentSummariesFor(application)
+            "artifacts" -> applicationService.getArtifactSummariesFor(application,
+              maxArtifactVersions ?: DEFAULT_MAX_ARTIFACT_VERSIONS)
+            else -> throw InvalidRequestException("Unknown entity type: $entity")
+          }
```

Comment:
the environments summary was pulled twice - once for `environments`, once for `artifacts`. Adding a function that takes care of the way deck is calling us so that we don't have to do the expensive database calculation twice. Leaving the old function in here for one-off calls, or when deck just calls with `resources`

Reply:
🥳 no idea how much you expect here in terms tangible impact on latency, but if you think it'll be material it'd be cool if we could watch a response latency graph of some kind before/after deployment

Reply:
ooooo. let me think.